scipy/stats/_distn_infrastructure.py

```diff
@@ -3836,3 +3884,423 @@ def get_distribution_names(namespace_pairs, rv_base_class):
         if isinstance(value, rv_base_class):
             distn_names.append(name)
     return distn_names, distn_gen_names
+
+
+def _combine_bounds(name, bounds_type, user_bounds, shape_domain,
+                    integral):
+    """Intersection of user-defined bounds and distribution PDF/PMF domain"""
```

Comment:
The user enters `bounds` for parameters, but also the distributions are only defined on specific domains (e.g. *p* in (0, 1)). This combines the information from both (and does input validation) before sending bounds to the optimizer.